code/game/machinery/constructable_frame.dm

```diff
@@ -229,10 +229,14 @@
 							// Move to nullspace and delete.
 							new_machine.circuit.moveToNullspace()
 							QDEL_NULL(new_machine.circuit)
-						for(var/obj/old_part in new_machine.component_parts)
-							// Move to nullspace and delete.
-							old_part.moveToNullspace()
-							qdel(old_part)
+						for(var/old_part in new_machine.component_parts)
+							// dont delete singleton datum parts
+							if(istype(old_part, /datum/stock_part))
+								continue
+							// Move to nullspace & delete other parts like cells, beakers etc.
+							var/obj/part = old_part
```

Comment:
This code was fine as is, the `obj` part already enforces a typecast.

Reply:
ohhhhhh. i'm dum